### DISCUSSION TITLE
Setting errno || Coverage of test cases || Existence of invalid fd case

### DISCUSSION THREAD

**Discussion by lagabulin** [ANLAB-KAIST/KENSv3#109] [Project Q&A 2. Transport Layer (TCP)] [answered]
Hi,

I got 3 questions about PA2.

1. Should I set the appropirate ERRNO when the corresponding situation occurs?
2. According to the lab session slide, we will need to pass "test-kens-all" binary. However, PA2 description file on KLMS required 4 binaries(open, bind, handshake and close). I would like to know which one will be graded.
![image](https://user-images.githubusercontent.com/111258004/192819341-11520dd7-d447-4cfa-98c3-88481f2a2bcb.png)
![image](https://user-images.githubusercontent.com/111258004/192819448-51c4da15-a296-4e2d-bf97-8797fc25a14b.png)
3. It might be a subquestion of the first one. Should we consider the case of invalid fd and handle it?

Thanks!

**Comment by whkim0** [comment, marked as answer]
1. Not necessarily.
2. Please refer to a post in classum.
3. I recommend to do so. You are okay unless you fail the test.

**Reply by lagabulin** [reply]
Thanks!
